hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/web/ozShell/acl/AclHandler.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.web.ozShell.acl;
+
+import org.apache.hadoop.ozone.client.OzoneClient;
+import org.apache.hadoop.ozone.security.acl.OzoneObj;
+import org.apache.hadoop.ozone.web.ozShell.Handler;
+import org.apache.hadoop.ozone.web.ozShell.OzoneAddress;
+
+import org.apache.hadoop.ozone.web.ozShell.StoreTypeOption;
+import picocli.CommandLine;
+
+import java.io.IOException;
+
+/**
+ * Base class for ACL-related commands.
+ */
+public abstract class AclHandler extends Handler {
```

Comment:
Based on child classes this class might implement the getAddress() method, as all of the descendants as I see implements that method exactly the same way. (returning address.getValue())

Reply:
They differ in the type of `address`, which can be `VolumeUri`, `BucketUri`, or `KeyUri`.

`AclHandler` subclasses mix two aspects: operation (get/set/etc.) and target (volume/bucket/key).  I kept the hierarchy shallow and extracted logic into `*Uri` and `AclOption`, so that duplicated implementations are reduced to one-liners.

We could introduce additional intermediate level classes into the hierarchy either by operation or target, but unfortunately the other aspect would still be duplicated.  I didn't find it worth introducing a new class to extract a one-liner method out of 4 classes.  (`extends ...` would still be duplicated ;) )

Reply:
Hmm, again you caught me with the different types... :)
Ok, so VolumeUri, BucketUri, and KeyUri implements "CommandLine.ITypeConverter<OzoneAddress>" this ensures the convert method from string to address which as I know is handled by PicoCLI.
If we introduce a common interface or any type of parent to deal with these URI implementations in the handlers then we can get around the multiple type problem in the ACLHandler, though I agree that does not worth it and probably does not help with understandability either. +1 to leave it this way, thank you for helping to understand it.